jupyter_client/channels.py

```diff
@@ -220,7 +220,7 @@ def _recv(self, **kwargs: t.Any) -> t.Dict[str, t.Any]:
         ident, smsg = self.session.feed_identities(msg)
         return self.session.deserialize(smsg)
 
-    def get_msg(self, timeout: t.Optional[float] = None) -> t.Dict[str, t.Any]:
+    def get_msg(self, timeout: t.Optional[int] = None) -> t.Dict[str, t.Any]:
```

Comment:
This should accept a float. `get_msg(timeout=0.25)` is valid. The cast to int should happen after multiplying by 1000 to get ms.